cpp/include/cudf/types.hpp

```diff
@@ -327,6 +328,21 @@ inline bool operator!=(data_type const& lhs, data_type const& rhs) { return !(lh
  */
 std::size_t size_of(data_type t);
 
+/**
+ * @brief Combines two hashed values into a single hashed value.
+ *
+ * Adapted from boost hash_combine, modified for 32/64-bit
+ * https://stackoverflow.com/a/4948967/1550940
+ * @param lhs The first hashed value
+ * @param rhs The second hashed value
+ * @return Combined hash value
+ */
+constexpr std::size_t hash_combine(std::size_t lhs, std::size_t rhs)
+{
+  constexpr std::size_t const magic = sizeof(std::size_t) == 8 ? 0x9e3779b97f4a7c15 : 0x9e3779b9;
```

Comment:
When is `sizeof(size_t) == 8` not true?

Reply:
`sizeof(size_t)=4`, in 32-bit systems.

Reply:
I don't think we have to worry about those anymore :)

Reply:
removed 32-bit.